src/providers/ldap/ldap_id.c

```diff
@@ -1917,6 +1920,23 @@ sdap_account_info_handler_send(TALLOC_CTX *mem_ctx,
         goto immediately;
     }
 
+#ifdef BUILD_SUBID
+    if ((data->entry_type & BE_REQ_TYPE_MASK) == BE_REQ_SUBID_RANGES) {
+        if (!id_ctx->opts->sdom->subid_ranges_search_bases ||
+            !id_ctx->opts->sdom->subid_ranges_search_bases[0] ||
+            !id_ctx->opts->sdom->subid_ranges_search_bases[0]->basedn) {
+            DEBUG(SSSDBG_OP_FAILURE, "subid_ranges_search_bases isn't set\n");
+            ret = EINVAL;
+            goto immediately;
+        }
+        data->extra_value = talloc_asprintf(data,
+            "%s=%s,"SYSDB_USERS_CONTAINER",%s",
+            id_ctx->opts->user_map[SDAP_AT_USER_NAME].name,
+            data->filter_value,
+            id_ctx->opts->sdom->user_search_bases[0]->basedn);
```

Comment:
Hi,

is there a reason you do not check for `NULL`? It is checked in `sdap_handle_acct_req_send()` but the related error message "This id_provider doesn't support subid ranges" might be irritating if talloc failed here.

bye,
Sumit

bye,
Sumit